src/components/gridlines.ts

```diff
@@ -4,31 +4,53 @@
  */
 
 import { Point, SimpleSelection } from "../core/interfaces";
-import { QuantitativeScale } from "../scales/quantitativeScale";
-import { IScaleCallback } from "../scales/scale";
+import { IScaleCallback, Scale } from "../scales/scale";
 
 import { Component } from "./component";
 
+function getScaledValue(
+  scale: Scale<any, any>,
+  between: boolean,
+  initialValue?: string) {
+
+  let previousPosition = initialValue === undefined
+    ? undefined
+    : scale.scale(initialValue);
+
+  return (tickVal: any) => {
+    const position = scale.scale(tickVal);
+
+    if (!between) {
+      return position;
+    }
+
+    let gridPosition: number;
+
+    if (previousPosition !== undefined) {
+      gridPosition = previousPosition + (position - previousPosition) / 2;
```

Comment:
furthermore, bar charts can also be aligned differently. in "histogram" mode the bar's left side is aligned to its x domain coordinate. this may be okay since you're passing the tick value through the scale, though.